urid/derive/src/uri_bound.rs

```diff
@@ -30,19 +28,21 @@ fn get_type_ident(item: TokenStream) -> Ident {
 ///
 /// This includes multiple checks to assure that the literal is formatted correctly.
 fn get_uri(attr: TokenStream) -> Literal {
-    lazy_static! {
-        static ref GET_STRING_RE: Regex = Regex::new(r#"^"(.*)"$"#).unwrap();
-    }
-
     const PARSING_ERROR: &str = "A URI has to be a string literal";
 
     if parse::<Literal>(attr.clone()).is_err() {
         panic!(PARSING_ERROR);
     }
-
+    //check if it's a litteral string
     let attr = attr.to_string();
-    let captures = GET_STRING_RE.captures(attr.as_str()).expect(PARSING_ERROR);
-    let uri = captures.get(1).expect(PARSING_ERROR).as_str();
+    if !attr.starts_with('"') || !attr.ends_with('"') {
+        panic!(PARSING_ERROR);
+    }
+    //Remove the enclosing "" to get the uri
+    let uri = String::from(attr.get(1..attr.len() - 1).expect(PARSING_ERROR));
+    if uri.contains(|c: char| c.is_ascii_whitespace()) {
+        panic!("A URI can't contain whitespace");
+    }
```

Comment:
Is it our job to judge whether the URI is actually a proper URI?

If so, we would need to do that properly and check for [all URI schemes](https://en.wikipedia.org/wiki/List_of_URI_schemes), which obviously is beyond our scope. Therefore, we should handle the "URIs" simply as strings and leave the validity up to the user. This also means that you can can also remove that ASCII check below, since there are non-ASCII URIs out there.

Reply:
ooops i resolved by error. Infact, i did that just because it was already panicking if the uri didn't contain a space because  ".*" doesn't capture space.

Reply:
Ok for the ascii check, i didn't know that uri can contain non-ascii char. I will remove it and replace it by a "\0" check